src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -74,14 +77,41 @@ public virtual void OpenStream(int streamIndex)
 
         protected override Task Send(int streamIndex, IList<Message> messages)
         {
-            return _connection.ScriptEvaluateAsync(
+            string GetMessageContent(Message message)
+            {
+                var content = Encoding.UTF8.GetString(message.Value.Array, message.Value.Offset, message.Value.Count);
+                return $"Key:{message.Key},ID:{message.MappingId},Content:{content}";
+            }
+
+            async Task WaitForAndTraceResult(Task<RedisResult> task)
+            {
+                var result = await task;
+                var values = (RedisValue[]) result;
+                _trace.TraceEvent(TraceEventType.Verbose, 0, $"Published message with key: {(int)values[0]}");
+            }
+
+            if (_trace.Switch.ShouldTrace(TraceEventType.Verbose))
+            {
+                var messageContent = "[" + string.Join(";", messages.Select(m => GetMessageContent(m))) + "]";
```

Comment:
Have you tested this under any kind of load? I'm guessing a stringbuilder would be more efficient, but I'm not sure whether it's necessary.

Reply:
Yeah, I guess we can remove it. This was focused on getting data for a specific scenario, not really something I planned to check in as a real change